### PR TITLE
Generate baselines by default and add replacement target

### LINKED ISSUE #410: Fix generate api check baselines task
- [ ] There is a target to run it as part of the build on a repository.
- [ ] There is a way to disable it though an MSBuild property (off by default).
- [ ] Baselines are generated for new assemblies.
- [ ] Breaking changes are deleted upon the generation of new baselines.

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Fixes #410.

@javiercn to check that this is what he intended.

### CODE REVIEW

File: docs/KoreBuild.md [hunk 1]
```diff
@@ -26,8 +26,10 @@ Below is a list of some of the properties that KoreBuild recognizes which you mi
 
 Property          | Purpose
 ------------------|--------
-GenerateBaselines | Toggles the (re-)generation of baselines for all projects.
+GenerateBaselines | Toggles the (re-)generation of baselines for all projects (defaults to true).
+ReplaceBaselines  | If 'true' baseline.*.json and breakingchanges.*.json files are deleted, and baseline.*.json files are recreated from the current state.
```
Comment (natemcmaster) [inline docs/KoreBuild.md]: image

Nit: escape asterisk with a backslash

File: src/ApiCheck.Console/Program.cs [hunk 2]
```diff
@@ -290,8 +290,16 @@ public static int Main(string[] args)
 
             foreach (var exclusion in incorrectBreakingChanges)
             {
-                Console.WriteLine(
-                    "ERROR: The following exclusion is in the exclusion file, but is no longer necessary:");
+                if (breakingChangesPathOption.HasValue())
+                {
+                    Console.WriteLine(
```
Comment (natemcmaster) [inline src/ApiCheck.Console/Program.cs]: Nit: Console.Error.WriteLine

File: src/Internal.AspNetCore.Sdk/buildMultiTargeting/ApiCheck.targets [hunk 3]
```diff
@@ -13,6 +13,15 @@
       RemoveProperties="TargetFrameworks" />
   </Target>
 
+  <Target
+      Name="Replace-ApiCheck-Baselines" >
+
+    <MSBuild Projects="$(MSBuildProjectFullPath)"
+      Targets="Replace-ApiCheck-Baselines"
+      Properties="TargetFramework=%(_TargetFrameworks.Identity)"
```
Comment (natemcmaster) [inline src/Internal.AspNetCore.Sdk/buildMultiTargeting/ApiCheck.targets]: You need to create the _TargetFrameworks item group. I may not be defined already.
```
     <ItemGroup>
       <_TargetFrameworks Remove="@(_TargetFrameworks)" />
       <_TargetFrameworks Include="$(TargetFrameworks)" />
     </ItemGroup>
```

File: src/Internal.AspNetCore.Sdk/build/ApiCheck.targets [hunk 4]
```diff
@@ -8,7 +8,8 @@
       <_ApiListingFileSuffix Condition=" '$(TargetFrameworkIdentifier)' == '.NETFramework' ">netframework.json</_ApiListingFileSuffix>
       <_ApiListingFileSuffix Condition=" '$(_ApiListingFileSuffix)' == '' ">netcore.json</_ApiListingFileSuffix>
       <_ApiListingFilePath>$(MSBuildProjectDirectory)\baseline.$(_ApiListingFileSuffix)</_ApiListingFilePath>
-      <_ApiListingFile Condition=" Exists('$(_ApiListingFilePath)') ">$(_ApiListingFilePath)</_ApiListingFile>
+      <_ApiListingOutputFile>$(MSBuildProjectDirectory)\bin\baseline.$(_ApiListingFileSuffix)</_ApiListingOutputFile>
```
Comment (natemcmaster) [inline src/Internal.AspNetCore.Sdk/build/ApiCheck.targets]: Bin directory may not be in the project directory. Better to use `$(OutputPath)`, not `$(MSBuildProjectDirectory)`. Test this on the aspnet/DotNetTools repo where the bin directory is no in the project dir. https://github.com/aspnet/DotNetTools/blob/dev/Directory.Build.props#L24

**Comment by ryanbrandenburg** [comment]
:up::date:

**Comment by javiercn** [comment]
Please port these changes to the 2.0.0 branch

**Pull request merged**